File: 1000-9999/5000/5543.cpp
// 상근날드
#include <bits/stdc++.h>
typedef long long ll;
using namespace std;

int burger[3];
int drink[2];
int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	cin >> burger[0] >> burger[1] >> burger[2];
	cin >> drink[0] >> drink[1];
	sort(burger, burger+3);
	sort(drink, drink + 2);
	cout << burger[0] + drink[0] - 50 << '\n';
}

File: 1000-9999/5000/5543.py
# 상근날드
burger = []
drink = []

for i in range(3):
    k = int(input())
    burger.append(k)
for i in range(2):
    k = int(input())
    drink.append(k)
print(min(burger) + min(drink) - 50)
